From 6fa665bc70d5f87282a3518595e39cc6cac99050 Mon Sep 17 00:00:00 2001
From: maximilian attems <max@stro.at>
Date: Fri, 5 Sep 2008 22:18:09 +0200
Subject: [PATCH 07/21] [klibc] mount: read /proc/mounts preferably

as klibc mount doesn't write into /etc/mtab (no addmntent() too),
better checkout the kernel view first.
may be interchanged again later..

Signed-off-by: maximilian attems <max@stro.at>
---
 usr/utils/mount_main.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/usr/utils/mount_main.c b/usr/utils/mount_main.c
index 89e75d7..ee08720 100644
--- a/usr/utils/mount_main.c
+++ b/usr/utils/mount_main.c
@@ -31,9 +31,9 @@ static __noreturn print_mount(char *type)
 	FILE *mfp;
 	struct mntent *mnt;
 
-	mfp = setmntent(_PATH_MOUNTED, "r");
+	mfp = setmntent(_PATH_PROC_MOUNTS, "r");
 	if (!mfp)
-		mfp = setmntent(_PATH_PROC_MOUNTS, "r");
+		mfp = setmntent(_PATH_MOUNTED, "r");
 	if (!mfp)
 		perror("setmntent");
 
-- 
1.6.5

